File: backend/src/main/java/com/manuelsch/simplekanban/service/TaskService.java
package com.manuelsch.simplekanban.service;

import com.manuelsch.simplekanban.DTOs.exceptionHandling.RecordNotFoundException;
import com.manuelsch.simplekanban.models.Priority;
import com.manuelsch.simplekanban.models.Task;

public interface TaskService {

    Task createTask(String boardColumnId, String title) throws RecordNotFoundException;

    Task updateTask(String id, String title, Priority priority, String color, String description) throws RecordNotFoundException;
}

File: frontend/src/main/web/src/app/services/task.service.ts
import { Injectable } from '@angular/core';
import { ApiService } from './api.service';
import { Task } from '../models/Task';
import { Priority } from '../models/Priority';

@Injectable({
  providedIn: 'root',
})
export class TaskService {

  constructor(private apiService: ApiService) {
  }

  public async createTask(boardColumnId: string, title: string): Promise<Task> {
    const body = {
      boardColumnId,
      title,
    };
    return this.apiService.post('/task', body);
  }

  public async updateTask(id: string, data: { title?: string, color?: string, priority?: Priority, description?: string }): Promise<Task> {
    const body = {
      id,
      ...data,
    };
    return this.apiService.post('/task/update', body);
  }
}
